### DISCUSSION TITLE
Front end: Proposal to create a “new data source” page

### DISCUSSION THREAD

**Discussion by gamab** [grafana/grafana#38924] [Data Sources]
# Context
I open this feature request as a follow up to our [slack discussion](https://raintank-corp.slack.com/archives/C025WTVRBSN/p1630583511036200) so that we can update the fine grained access control front end permissions, when/if this page is created. :pray: 

# Description
I've been working on protecting data sources related endpoints with fine grained access control (ex: users with `datasources:read` rights can target `GET /api/datasources/`). This was part of the PR #38070.
While I was working on this, I noticed that, in the `/datasources/new` page, when we select a data source (ex: influxdb) it gets created **straight away** with default values and **then** the end user is redirected to the edit page of this newly created data source. 

# Suggestion
An optimal experience would be to present end users with a `new data source` page in which they'd be able to configure the data source settings **before** the actual creation. (@ashharrison90 hope you don't mind me reusing your wording :slightly_smiling_face:)

Regarding access control, my concern was that, if users have `create` rights without `write` rights, the current state of the front-end code would allow them to create a data source with default values but would not allow them to edit it.

Other users have mentioned that the absence of an extra config page sometimes lead them to create, by mistake, blank data sources that they would have to remove straight away.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
